app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,230 @@
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: SessionControlInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+
+    private lateinit var messageGlobal: Message
 
     fun bind(message: Message) {
-        val binding = NimbusMessageCardBinding.bind(itemView)
+        messageGlobal = message
+        interactor.onMessageDisplayed(message)
+    }
 
-        if (message.data.title.isNullOrBlank()) {
-            binding.titleText.isVisible = false
-        } else {
-            binding.titleText.text = message.data.title
-        }
+    companion object {
+        internal val LAYOUT_ID = View.generateViewId()
+    }
 
-        binding.descriptionText.text = message.data.text
+    @Composable
+    override fun Content() {
 
-        if (message.data.buttonLabel.isNullOrBlank()) {
-            binding.messageButton.isVisible = false
-            binding.experimentCard.setOnClickListener {
-                interactor.onMessageClicked(message)
-            }
-        } else {
-            binding.messageButton.text = message.data.buttonLabel
-            binding.messageButton.setOnClickListener {
-                interactor.onMessageClicked(message)
+        MessageCardPlaceHolder(
+            message = messageGlobal,
+            onMessageClicked = { interactor.onMessageClicked(messageGlobal) },
+            onMessageClosedClicked = { interactor.onMessageClosedClicked(messageGlobal) }
+        )
+    }
+}
+
+/**
+ * Message Card.
+ *
+ * @param message contains title and description of message.
+ * @param onMessageClicked Invoked when user clicks on the button.
+ * @param onMessageClosedClicked Invoked when user clicks on close button to remove message.
+ */
+@Suppress("LongMethod")
+@Composable
+fun MessageCardPlaceHolder(
+    message: Message,
+    onMessageClicked: () -> Unit,
+    onMessageClosedClicked: () -> Unit,
+) {
+    Card(
+        modifier = Modifier
+            .padding(16.dp)
+            .then(
+                if (message.data.buttonLabel.isNullOrBlank()) {
+                    Modifier.clickable(onClick = onMessageClicked)
+                } else {
+                    Modifier
+                }
+            ),
+        shape = RoundedCornerShape(8.dp),
+        backgroundColor = FirefoxTheme.colors.layer2,
+    ) {
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            if (!message.data.title.isNullOrBlank()) {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                ) {
+
+                    SectionHeader(
+                        text = message.data.title!!,
+                        modifier = Modifier
+                            .weight(1f)
+                            .padding(bottom = 10.dp)
+                    )
+
+                    IconButton(
+                        modifier = Modifier.size(20.dp),
```

Comment:
Should this be 48dp be meet the [minimal touch size for accessibility](https://support.google.com/accessibility/android/answer/7101858?hl=en)? :)

Reply:
Looking at the internals of `IconButton`, it seems that regardless of the size specified, it will still have a 48x48 tap target size. During IconButton's setup, it calls `.minimumTouchTargetSize()` on its root `Modifier`

Reply:
Awesome 🙌🏽 !